central/declarativeconfig/updater/access_scope_updater_test.go

```diff
@@ -0,0 +1,216 @@
+//go:build sql_integration
+
+package updater
+
+import (
+	"context"
+	"testing"
+
+	declarativeConfigHealth "github.com/stackrox/rox/central/declarativeconfig/health/datastore"
+	roleDS "github.com/stackrox/rox/central/role/datastore"
+	roleMocks "github.com/stackrox/rox/central/role/datastore/mocks"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+func TestAccessScopeUpdater(t *testing.T) {
+	suite.Run(t, new(updaterTestSuite))
+}
+
+type updaterTestSuite struct {
+	suite.Suite
+
+	ctx     context.Context
+	pgTest  *pgtest.TestPostgres
+	updater *accessScopeUpdater
+}
+
+func (s *updaterTestSuite) SetupTest() {
+	s.ctx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Access, resources.Integration),
+		),
+	)
+	s.ctx = declarativeconfig.WithModifyDeclarativeOrImperative(s.ctx)
+
+	s.pgTest = pgtest.ForT(s.T())
+	s.Require().NotNil(s.pgTest)
+	ds, err := roleDS.GetTestPostgresDataStore(s.T(), s.pgTest.DB)
+	s.Require().NoError(err)
+	s.updater = newAccessScopeUpdater(ds,
+		declarativeConfigHealth.GetTestPostgresDataStore(s.T(), s.pgTest.DB)).(*accessScopeUpdater)
+}
+
+func (s *updaterTestSuite) TearDownTest() {
+	s.pgTest.Teardown(s.T())
+	s.pgTest.Close()
+}
+
+func (s *updaterTestSuite) TestUpsert() {
+	cases := map[string]struct {
+		m   protocompat.Message
+		err error
+	}{
+		"invalid message type should yield an error": {
+			m:   &storage.Role{Name: "something"},
+			err: errox.InvariantViolation,
+		},
+		"valid message type should be upserted": {
+			m: &storage.SimpleAccessScope{
+				Id:          "61a68f2a-2599-5a9f-a98a-8fc83e2c06cf",
+				Name:        "testing",
+				Description: "testing",
+				Rules: &storage.SimpleAccessScope_Rules{
+					IncludedClusters: []string{"cluster1"},
+				},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		s.Run(name, func() {
+			err := s.updater.Upsert(s.ctx, tc.m)
+			s.ErrorIs(err, tc.err)
+			if tc.err == nil {
+				_, exists, err := s.updater.roleDS.GetAccessScope(s.ctx, s.updater.idExtractor(tc.m))
+				s.NoError(err)
+				s.True(exists)
+			}
+		})
+	}
+}
+
+func (s *updaterTestSuite) TestDelete_Successful() {
+	scopes := []*storage.SimpleAccessScope{
+		{
+			Id:          "61a68f2a-2599-5a9f-a98a-8fc83e2c06cf",
```

Comment:
How about ids that are easier to remember
```suggestion
			Id:          "11111-....",
```